### PR TITLE
242 defaultcelo

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description

Supporting changes for defaulting protocolUI to CELO.
- load balances from sdkv2
- load native balances from sdkv2
- refactor savings to celo

About # (link your issue here)
https://github.com/GoodDollar/GoodProtocolUI/issues/242

# Todo
- [x] load gdx for mainnet (useG$Balances)
- [x] fix bug on staging env / fuse network (see code-comment) (savings hooks)

# How Has This Been Tested?

locally, web.

### CODE REVIEW

File: packages/sdk-v2/src/sdk/savings/react.ts [hunk 1]
```diff
@@ -102,16 +101,16 @@ export const useSavingsFunctions = () => {
...
+  const gdStaking = useGetContract("GoodDollarStaking", true, "savings", chainId) as GoodDollarStaking;
 
-  const gdStaking = useGetContract(
-    "GoodDollarStaking",
-    true,
-    "savings",
-    defaultEnv,
-    SupportedChains.FUSE
-  ) as GoodDollarStaking;
+  // const stats = gdStaking.stats(); // todo-fix: weird behaviour on staging env and connected to fuse network.
```
Comment (L03TJ3) [inline packages/sdk-v2/src/sdk/savings/react.ts:108]: This I cannot figure out

File: packages/sdk-v2/src/sdk/savings/sdk.ts [hunk 2]
```diff
@@ -1,66 +1,65 @@
...
-        return res
-      })
-      return req
-    } catch (e) { console.log('get staker info failed', {e})}
+      const req = await contract.stakersInfo(account).then(res => {
+        return res;
+      });
+      return req;
+    } catch (e) {
+      console.log("get staker info failed", { e });
```
Comment (sirpy) [inline packages/sdk-v2/src/sdk/savings/sdk.ts:62]: dont have an empty catch, either remove it or return something meaningful

**Review by sirpy — CHANGES_REQUESTED** [review]

File: packages/sdk-v2/src/hooks/useNativeBalance.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import { useEtherBalance, useEthers } from "@usedapp/core";
+import { formatEther } from "ethers/lib/utils";
+export const useNativeBalance = () => {
+  const { account } = useEthers();
+  const nativeBalance = useEtherBalance(account);
+  if (nativeBalance) {
+    const formattedBalance = formatEther(nativeBalance);
+    return parseFloat(formattedBalance).toFixed(6);
```
[the remark this answers is not in the document]
Comment (johnsmith-gooddollar) [reply packages/sdk-v2/src/hooks/useNativeBalance.tsx:9]: or at least add format opts (e.g. format string) as hook parameter

File: packages/sdk-v2/src/sdk/base/react.ts [hunk 4]
```diff
@@ -26,15 +29,16 @@ export const useReadOnlySDK = (type: SdkTypes, requiredChainId?: number): Reques
   return useSDK(true, type, requiredChainId);
 };
 
-export const useGetEnvChainId = (requiredChainId?: number) => {
+export const useGetEnvChainId = (requiredChainId?: number, v2Supported?: (string | SupportedV2Networks)[]) => {
   const { chainId } = useEthers();
   const web3Context = useContext(Web3Context);
   let baseEnv = web3Context.env || "";
   let connectedEnv = baseEnv;
+  const v2ChainId = v2Supported?.includes(chainId as SupportedV2Networks) ? chainId : SupportedChains.CELO;
```
[the remark this answers is not in the document]
Comment (johnsmith-gooddollar) [reply packages/sdk-v2/src/sdk/base/react.ts:37]: looks like fallback to Celo if the wrong chain been set

File: packages/sdk-v2/src/sdk/base/react.ts [hunk 5]
```diff
@@ -26,15 +29,16 @@ export const useReadOnlySDK = (type: SdkTypes, requiredChainId?: number): Reques
...
 
-  switch (requiredChainId ?? chainId) {
+  switch (v2Supported ? v2ChainId : requiredChainId ?? chainId) {
     case 1:
-      connectedEnv += "-mainnet";
+      "production-mainnet"; // temp untill dev contracts are released to goerli
```
[the remark this answers is not in the document]
Comment (johnsmith-gooddollar) [reply packages/sdk-v2/src/sdk/base/react.ts:41]: if its dev temporary fix please add if env=dev condition

File: packages/sdk-v2/src/sdk/base/react.ts [hunk 6]
```diff
@@ -126,3 +129,86 @@ export const useSDK = (
...
+    };
+
+    // const gdxBalance = {
+    //   amount: CurrencyValue.fromString(gdx, results[2]?.value[0].toString()),
+    //   token: new Currency("G$X", "GDX", 2)
+    // };
+
+    return { g$balance, goodBalance, gdxBalance: undefined };
+  } else {
+    return { g$Balance: undefined, goodBalance: undefined, gdxBalance: undefined };
```
Comment (johnsmith-gooddollar) [inline packages/sdk-v2/src/sdk/base/react.ts:212]: the same

```suggestion
    return { }; 
```

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: packages/sdk-v2/src/sdk/savings/sdk.ts (hunk 2 above)
Comment (L03TJ3) [reply packages/sdk-v2/src/sdk/savings/sdk.ts]: catch cancellation now, when merged we later might want to log some error to sentry?

File: packages/sdk-v2/src/sdk/savings/sdk.ts [hunk 7]
```diff
@@ -1,66 +1,65 @@
...
+    try {
+      //note: if tx fails on limit, up the gasLimitBufferPercentage (see context config))
+      let shares = isFullWithdraw
+        ? await contract.sharesOf(await contract.signer.getAddress())
+        : await contract.amountToShares(amount);
+      const withdraw = contract.withdrawStake(shares);
+      return withdraw;
+    } catch (e) {
+      console.log("withdraw savings failed -->", e);
+      return new Error(e as any);
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/sdk-v2/src/sdk/savings/sdk.ts]: kept it to catch cancellation by user

File: packages/sdk-v2/src/hooks/useNativeBalance.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/sdk-v2/src/hooks/useNativeBalance.tsx]: formatting moved to UI

File: packages/sdk-v2/src/sdk/base/react.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/sdk-v2/src/sdk/base/react.ts]: Not needed, using required chain now from UI > sdk

File: packages/sdk-v2/src/sdk/savings/react.ts [hunk 8]
```diff
@@ -102,16 +101,16 @@ export const useSavingsFunctions = () => {
 
 export const useSavingsStats = (refresh: QueryParams["refresh"] = "never") => {
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/sdk-v2/src/sdk/savings/react.ts]: done

File: packages/sdk-v2/src/sdk/savings/react.ts [hunk 9]
```diff
@@ -102,16 +101,16 @@ export const useSavingsFunctions = () => {
...
   const refreshOrNever = useRefreshOrNever(refresh);
+  const { chainId, defaultEnv } = useGetEnvChainId(undefined, Object.values(SupportedV2Networks));
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/sdk-v2/src/sdk/savings/react.ts]: no it wasnt

File: packages/sdk-v2/src/sdk/savings/sdk.ts [hunk 10]
```diff
@@ -1,66 +1,65 @@
...
 
-      const transfer = contract.transferAndCall(stakeContract.address, amount, callData)
-      return transfer
+      const transfer = contract.transferAndCall(stakeContract.address, amount, callData);
+      return transfer;
     } catch (e) {
-      console.log('onTokenTransfer failed -->', {e}) 
-      return new Error(e as any)
+      console.log("onTokenTransfer failed -->", { e });
+      return new Error(e as any);
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/sdk-v2/src/sdk/savings/sdk.ts]: changed

File: packages/sdk-v2/src/sdk/base/react.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/sdk-v2/src/sdk/base/react.ts]: forgot to set connectedEnv. 
but mainly this ensures (temporarily) that when on dev envs, ropsten/kovan contracts are not being loaded

File: packages/sdk-v2/src/sdk/base/react.ts [hunk 11]
```diff
@@ -126,3 +129,86 @@ export const useSDK = (
...
   return sdk;
 };
+
+/* Not sure about placement of this hook, use base for now */
+
+export function useG$Tokens() {
+  const { chainId, defaultEnv, baseEnv } = useGetEnvChainId();
+  const g$ = G$ContractAddresses("GoodDollar", defaultEnv) as string;
+  const good = G$ContractAddresses("GReputation", defaultEnv) as string;
+  const gdx = G$ContractAddresses("GoodReserveCDai", baseEnv + "-mainnet") as string;
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/sdk-v2/src/sdk/base/react.ts:138]: works

File: packages/sdk-v2/src/sdk/savings/sdk.ts [hunk 12]
```diff
@@ -1,66 +1,65 @@
...
-    const contract = this.getContract("GoodDollarStaking")
+  async hasBalance(account: string): Promise<boolean | undefined> {
+    const contract = this.getContract("GoodDollarStaking");
     if (contract && account) {
-      const balance = (await contract.balanceOf(account)).toString()
-      const hasBalance = balance !== '0'
-      return hasBalance
+      const balance = (await contract.balanceOf(account)).toString();
+      const hasBalance = balance !== "0";
+      return hasBalance;
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/sdk-v2/src/sdk/savings/sdk.ts]: using bn now

**Comment by johnsmith-gooddollar** [comment]
@L03TJ3 why closed ? will you open new PR to merge ?

**Comment by L03TJ3** [comment]
I squashed merged, isn't that what you requested before?

**Comment by johnsmith-gooddollar** [comment]
@L03TJ3 I see you've **closed** it
without merging

**Comment by johnsmith-gooddollar** [comment]
ok I see your commits in master. let's discuss later 
maybe you did it manually, but actually you had to chose 'squash and merge' from GitHub UI

**Pull request closed without merging**